backend/routers/routes/courses.py

```diff
@@ -1,9 +1,98 @@
 from crud import user_crud
-from fastapi import APIRouter, Depends, HTTPException, Request, Response
+from fastapi import APIRouter, HTTPException, Request, Response
+from typing import List, Optional
+from pydantic import BaseModel, Field, validator
+
+from core.pipelines.doc_query_engine import DocumentQueryEngine
 
 course_router = APIRouter()
 
 
+# Request model
+class CourseQueryRequest(BaseModel):
+    question: str = Field(..., min_length=5)
+    template_name: Optional[str] = None
+    max_chunks: Optional[int] = Field(default=5, ge=1)
+
+    @validator("question")
+    def question_length(cls, v):
+        if len(v.strip()) < 5:
+            raise ValueError("Question must be at least 5 characters long")
+        return v
+
+
+# Response model
+class Source(BaseModel):
+    document_title: str
+    content: str
+    similarity: float
+    metadata: dict
+
+
+class CourseQueryResponse(BaseModel):
+    answer: str
+    sources: List[Source]
+
+
+def user_has_access_to_course(user_id: Optional[str], course_id: str) -> bool:
+    # Implement your logic to check if the user has access to the course
+    # For example, check if user is enrolled in the course
+    # Return True if they have access, False otherwise
+    # For now, we'll assume all users have access (replace with real logic)
+    return True
+
+
+@course_router.post("/courses/{course_id}/query", response_model=CourseQueryResponse)
+async def query_course_content(
+    course_id: str, query_request: CourseQueryRequest, request: Request
+):
+    """
+    Endpoint for querying course content using RAG.
+    """
+    user_id = request.headers.get("X-User-ID")
+
+    if not user_has_access_to_course(user_id, course_id):
+        raise HTTPException(status_code=403, detail="Access denied to this course")
+
+    # Initialize DocumentQueryEngine
+    try:
+        query_engine = DocumentQueryEngine(max_chunks=query_request.max_chunks)
+    except Exception as e:
+        raise HTTPException(
+            status_code=500, detail=f"Error initializing query engine: {e}"
+        )
```

Comment:
I think this should be initialized outside the post endpoint. it doesn't need to be init'd for every request.